test/units/modules/network/dcnm/test_dcnm_vrf.py

```diff
@@ -0,0 +1,99 @@
+# (c) 2020 Red Hat Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from units.compat.mock import patch
+from ansible.modules.network.dcnm import dcnm_vrf
+from .dcnm_module import TestDcnmModule, set_module_args
+
+
+class TestDcnmVrfModule(TestDcnmModule):
+
+    module = dcnm_vrf
+
+    def setUp(self):
+        super(TestDcnmVrfModule, self).setUp()
+
+        self.mock_dcnm_connection = patch('ansible.modules.network.dcnm.dcnm_vrf.dcnm_connection')
+        self.run_dcnm_connection = self.mock_dcnm_connection.start()
+
+    def tearDown(self):
+        super(TestDcnmVrfModule, self).tearDown()
+        self.mock_dcnm_connection.stop()
+
+    def load_fixtures(self, response=None, device=''):
+        create_err_resp = list([dict(ERROR='Bad Request')])
+        attach_success_resp = dict({'vrf-on-switch':'SUCCESS'})
+        attach_err_resp = dict({'vrf-on-switch':'Entered VRF VLAN ID 103 is in use already'})
+        deploy_success_resp = dict({"status":""})
+
+        if '_create_new' in self._testMethodName:
+            self.run_dcnm_connection.side_effect = [create_err_resp, dict()]
+        elif '_create_duplicate' in self._testMethodName:
+            self.run_dcnm_connection.side_effect = [dict(vrfName='test_vrf'), dict()]
+        elif '_create_with_used' in self._testMethodName:
+            self.run_dcnm_connection.side_effect = [create_err_resp, create_err_resp]
+        elif 'attach_multiple' in self._testMethodName or 'attach_single' in self._testMethodName:
+            self.run_dcnm_connection.side_effect = [attach_success_resp]
+        elif 'used_vlan' in self._testMethodName:
+            self.run_dcnm_connection.side_effect = [attach_err_resp]
+        elif 'deploy' in self._testMethodName:
+            self.run_dcnm_connection.side_effect = [deploy_success_resp]
+        else:
+            pass
+
+    def test_dcnm_vrf_create_new(self):
```

Comment:
So if I am reading this correctly you are mocking two calls to run_dcnm_connection.

1. The first time DCNM would return `list([dict(ERROR='Bad Request')])` because the vrf does not exist.
2. The second time DCNM would return an empty `dict()` to indicate success?

Is this all of the information we are getting back from DCNM in the case were we are really interacting with DCNM?

Reply:
Yes, mocking dcnm_connection twice,
1. Returns "ERROR: Bad request" as one of the few other key:value pairs, but, in the response handling code in the module, I am only looking for list[0].dict('ERROR'). Hence, the payload in the mock is limited.
2. Yes, just empty dict to indicate success.